hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -362,13 +370,18 @@ public HoodieLogBlock prev() throws IOException {
     // blocksize should read everything about a block including the length as well
     try {
       inputStream.seek(reverseLogFilePosition - blockSize);
+      // get the block size from head and match it with the block size from tail
```

Comment:
This code is pretty complicated to understand. I see that you removed hasNext() and added some code around this + handling of corrupt blocks here. Can this be simplified with hasNext(), checkCorruptBlock() method level abstractions ? We need more logs as well to explain such checks..

Reply:
talked offline. will create a new JIRA to address reverse reading issues